### PR TITLE
Shopify: create checkout on add to cart

### PR DESCRIPTION
**Pull request by cond0r**
Creates a checkout only when user first time interacts with the cart, not on the first visit of the store #431

### CODE REVIEW

**Comment by ottomated** [comment]
This is promising but I would say it needs both an option to override this behavior and for this to happen on any cart-related action.

**Comment by cond0r** [comment]
> This is promising but I would say it needs both an option to override this behavior and for this to happen on any cart-related action.

The other cart related  are not active since the cart is empty / null. So you cannot click delete or update. And i just checked. I deleted the cart cookie, and you click on cart again, and the cart was empty, and i was able to click on add to cart, and the cookie & checkout was created again.

File: framework/bigcommerce/cart/use-add-item.tsx [hunk 1]
```diff
@@ -29,16 +29,18 @@ export const handler: MutationHook<AddItemHook> = {
 
     return data
   },
-  useHook: ({ fetch }) => () => {
-    const { mutate } = useCart()
+  useHook:
```
Comment (goncy) [inline framework/bigcommerce/cart/use-add-item.tsx:32]: Can you revert the changes for this file?

File: tsconfig.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
       "@components/*": ["components/*"],
       "@commerce": ["framework/commerce"],
       "@commerce/*": ["framework/commerce/*"],
-      "@framework": ["framework/local"],
-      "@framework/*": ["framework/local/*"]
+      "@framework": ["framework/shopify"],
```
Comment (goncy) [inline tsconfig.json:26]: Can you revert this changes?

**Review by goncy — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 2 above)
Comment (cond0r) [reply tsconfig.json]: @goncy done

**Review by lfades — CHANGES_REQUESTED** [review]
Hi there and thank you for the PR 👋 

Changes look good to me but the checkout isn't working in the demo: https://commerce-shopify-l41k21i4l-vercel-solutions-vtest314.vercel.app/

**Comment by cond0r** [comment]
> Hi there and thank you for the PR 👋
> 
> Changes look good to me but the checkout isn't working in the demo: https://commerce-shopify-l41k21i4l-vercel-solutions-vtest314.vercel.app/

I think it as something to do with preview URL's you can see it did the same for the Implement custom checkout one https://commerce-shopify-git-implement-bfb334-vercel-solutions-vtest314.vercel.app/  but is ok in production one

**Comment by cond0r** [comment]
@lfades now should work. On #487 the handler for checkout was changed to be on submit-checkout file which happens on POST, but since the /checkout is a redirect to /api/checkout, it has GET headers, so changed it to be on get-checkout handler. But what is strange, is that also BigCommerce should require the same changes to work, but works in the production URL now (same with Shopify). But it doesn't work on the preview URL's from the PR for example BigCommerce https://commerce-bigcommerce-git-imple-57f06f-vercel-solutions-vtest314.vercel.app/checkout

**Comment by goncy** [comment]
> @lfades now should work. On #487 the handler for checkout was changed to be on submit-checkout file which happens on POST, but since the /checkout is a redirect to /api/checkout, it has GET headers, so changed it to be on get-checkout handler. But what is strange, is that also BigCommerce should require the same changes to work, but works in the production URL now (same with Shopify). But it doesn't work on the preview URL's from the PR for example BigCommerce https://commerce-bigcommerce-git-imple-57f06f-vercel-solutions-vtest314.vercel.app/checkout

You are right @cond0r, will take a look at this and get back at you so we can merge this 🙌

**Comment by goncy** [comment]
@cond0r a PR has been merged to solve this, you can get back to use get-checkout as submit-checkout is not required anymore, thanks! Let me know once its done and I can merge this 🙌

**Comment by cond0r** [comment]
> @cond0r a PR has been merged to solve this, you can get back to use get-checkout as submit-checkout is not required anymore, thanks! Let me know once its done and I can merge this 🙌

@goncy done, you can check again

**Review by goncy — APPROVED** [review]

**Comment by goncy** [comment]
Merged, thanks @cond0r!

**Pull request merged**